tests/components/ring/test_light.py

```diff
@@ -90,3 +95,40 @@ async def test_updates_work(
 
     state = hass.states.get("light.front_light")
     assert state.state == "on"
+
+
+@pytest.mark.parametrize(
+    ("exception_type", "reauth_expected"),
+    [
+        (ring_doorbell.AuthenticationError, True),
+        (ring_doorbell.RingTimeout, False),
+        (ring_doorbell.RingError, False),
+    ],
+    ids=["Authentication", "Timeout", "Other"],
+)
+async def test_light_errors_when_turned_on(
+    hass: HomeAssistant,
+    requests_mock: requests_mock.Mocker,
+    exception_type,
+    reauth_expected,
+) -> None:
+    """Tests the light turns on correctly."""
+    await setup_platform(hass, Platform.LIGHT)
+    config_entry = hass.config_entries.async_entries("ring")[0]
+
+    assert not any(config_entry.async_get_active_flows(hass, {SOURCE_REAUTH}))
+
+    with patch.object(
+        ring_doorbell.RingStickUpCam, "lights", new_callable=PropertyMock
+    ) as mock_lights:
+        mock_lights.side_effect = exception_type
+        with pytest.raises(HomeAssistantError):
+            await hass.services.async_call(
+                "light", "turn_on", {"entity_id": "light.front_light"}, blocking=True
+            )
+        await hass.async_block_till_done()
+    assert mock_lights.call_count == 1
+    assert (
```

Comment:
For completeness check that the reuath flow is for this domain

Reply:
Now done